transport/interface.go

```diff
@@ -0,0 +1,18 @@
+package transport
+
+import (
+	"context"
+
+	"github.com/libp2p/go-libp2p-core/event"
+
+	"github.com/filecoin-project/boost/transport/types"
+)
+
+type Transport interface {
+	Execute(ctx context.Context, transportInfo []byte, dealInfo *types.TransportDealInfo) (th Handler, isComplete bool, err error)
```

Comment:
can we get comments through this interface. it's hard to understand exactly how to plug in other implementations given this interface.